### ISSUE TITLE
Can't navigate to errors in haxelibs ("unable to open file")

### ISSUE THREAD

**Issue by Gama11** [vshaxe/vshaxe#23]
When an error occurs in a haxelib, the path is absolute, breaking navigation with the problems view:

![image](https://cloud.githubusercontent.com/assets/2620907/17984971/ee3f1074-6b13-11e6-8754-a06580e5f587.png)

The problem matcher assumes relative paths. `"fileLocation": "absolute"` helps, but it breaks other errors with relative filepaths.

Waiting on https://github.com/Microsoft/vscode/issues/449 to be resolved.

**Comment by Gama11** [comment]
There's another issue here, and that's that even if `"fileLocation": ["autoDetect"]` is supported by VSCode, it would still be relative to the current VSCode workspace.

This wouldn't work for vshaxe itself, for instance, because we have nested projects and "cascading" `build.hxml` files (for instance, the root `build.hxml` switches to `--cwd server` and runs the `build.hxml`) there.

The best solution would be just getting Haxe to emit absolute paths... Then again, we might be able to work around that if we control the build process (#79).

**Comment by ncannasse** [comment]
I think we should start implementing a -D vscode in Haxe, that might activate several features such as 1-based columns, absolute paths, etc. It would be required for now to pass it manually in tasks.json, then we can soon have it automatically passed.

@Simn what do you think?

**Comment by EricBishton** [comment]
Are you going to add -D HaxeDevelop, -D IDEA, -D SublimeText, -D vim... as well?  In my opinion, it's not a good thing to add bits of functionality to the compiler that have to be supported for each potential environment.

If you really want to add extra code paths to deal with niggling issues, make sure that they are all independently available  (e.g. -D absolutePaths, -D zeroBasedColumns) and not gathered together under an umbrella define (e.g. -D vscode).  It's not a problem for an IDE to pass a large number of defines.  It is a problem to have them adjust to a number of behavioral changes because they need one particular fix that's under an umbrella.

**Comment by nadako** [comment]
Yeah, Haxe shouldn't need to know anything about IDEs.

**Comment by ncannasse** [comment]
For HL debugger I needed to map the trace() paths from classpath-relative to absolute, it's actually very little code to read the HXML and build the classpath:
https://github.com/ncannasse/vscode_hl_debug/blob/nc/src/HLAdapter.hx#L81

Another option would be indeed to ask haxe to print absolute path errors. We have -D absolute-path, but it's only for trace() so we need another one.

**Comment by ncannasse** [comment]
https://github.com/Microsoft/vscode/issues/449 has been opened for more than two years and there is no timeline for it. I think we should definitely move forward with our own solution like I did in HL vscode integration, it's very painful to work on a project when you want to be able to modify the local git installed libraries at the same time.
